### PR TITLE
added GetCacheEntryRequest for MC cache browser

### PR DESCRIPTION
**Pull request by hasancelik**

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/cache/impl/CacheEntryProcessorEntry.java [hunk 1]
```diff
@@ -141,6 +141,10 @@ protected V getRecordValue(R record) {
         return (V) objValue;
     }
 
+    public R getRecord() {
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/cache/impl/CacheEntryProcessorEntry.java]: maybe we can add an assertion here to be sure that this method is called by partition threads `assert !(Thread.currentThread() instanceof PartitionOperationThread)`

File: hazelcast/src/main/java/com/hazelcast/internal/management/request/GetCacheEntryRequest.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+    private static class GetCacheEntryViewEntryProcessor implements EntryProcessor<Object, Object, CacheEntryView> {
+        @Override
+        public CacheEntryView process(MutableEntry mutableEntry, Object... objects) throws EntryProcessorException {
+            final CacheEntryProcessorEntry entry = (CacheEntryProcessorEntry) mutableEntry;
+            final CacheRecord record = entry.getRecord();
+            CacheEntryView<Object, Object> cacheEntryView = new CacheEntryView<Object, Object>() {
+                @Override
+                public String getKey() {
+                    return null;
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/internal/management/request/GetCacheEntryRequest.java]: a comment can be useful to see why we return `null` here

File: hazelcast/src/main/java/com/hazelcast/internal/management/request/GetCacheEntryRequest.java [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+    @Override
+    public void writeResponse(ManagementCenterService mcs, JsonObject root) throws Exception {
+        InternalSerializationService serializationService = mcs.getHazelcastInstance().getSerializationService();
+        HazelcastInstanceCacheManager cacheManager = mcs.getHazelcastInstance().getCacheManager();
+        ICache<Object, Object> cache = cacheManager.getCache(cacheName);
+        GetCacheEntryViewEntryProcessor entryProcessor = new GetCacheEntryViewEntryProcessor();
+        CacheEntryView cacheEntry = null;
+
+        if (type.equals("string")) {
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/internal/management/request/GetCacheEntryRequest.java]: better to write this `"string".equals(type)` to be safe from NPE

**Comment by ahmetmircik** [comment]
@hasancelik can we also add tests for different storage types (especially for NATIVE)?

File: hazelcast/src/main/java/com/hazelcast/internal/management/request/GetCacheEntryRequest.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        }
+        return properties;
+    }
+
+    @Override
+    public void writeResponse(ManagementCenterService mcs, JsonObject root) throws Exception {
+        InternalSerializationService serializationService = mcs.getHazelcastInstance().getSerializationService();
+        HazelcastInstanceCacheManager cacheManager = mcs.getHazelcastInstance().getCacheManager();
+        ICache<Object, Object> cache = cacheManager.getCache(cacheName);
+        GetCacheEntryViewEntryProcessor entryProcessor = new GetCacheEntryViewEntryProcessor();
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/internal/management/request/GetCacheEntryRequest.java]: `entryProcessor` can be a constant

File: hazelcast/src/main/java/com/hazelcast/cache/impl/CacheEntryProcessorEntry.java (hunk 1 above)
Comment (hasancelik) [inline hazelcast/src/main/java/com/hazelcast/cache/impl/CacheEntryProcessorEntry.java]: ✅

File: hazelcast/src/main/java/com/hazelcast/internal/management/request/GetCacheEntryRequest.java (hunk 2 above)
Comment (hasancelik) [inline hazelcast/src/main/java/com/hazelcast/internal/management/request/GetCacheEntryRequest.java]: ✅

File: hazelcast/src/main/java/com/hazelcast/internal/management/request/GetCacheEntryRequest.java (hunk 3 above)
Comment (hasancelik) [inline hazelcast/src/main/java/com/hazelcast/internal/management/request/GetCacheEntryRequest.java]: ✅

File: hazelcast/src/main/java/com/hazelcast/internal/management/request/GetCacheEntryRequest.java (hunk 4 above)
Comment (hasancelik) [inline hazelcast/src/main/java/com/hazelcast/internal/management/request/GetCacheEntryRequest.java]: ✅

**Comment by hasancelik** [comment]
@ahmetmircik added tests for `GetCacheEntryRequestTest` and `GetMapEntryRequestTest`

File: hazelcast/src/test/java/com/hazelcast/internal/management/GetMapEntryRequestTest.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    @Test
+    public void testGetMapEntry() throws Exception {
+        GetMapEntryRequest request = new GetMapEntryRequest("string","map","key");
+        IMap<String, String> map = hz.getMap("map");
+        map.put("key","value");
+
+        JsonObject jsonObject = new JsonObject();
+        request.writeResponse(managementCenterService, jsonObject);
+        JsonObject result = (JsonObject) jsonObject.get("result");
+        System.out.println(result);
```
Comment (ahmetmircik) [inline hazelcast/src/test/java/com/hazelcast/internal/management/GetMapEntryRequestTest.java]: `System.out.println(result)` seems left over?

File: hazelcast/src/main/java/com/hazelcast/internal/management/request/GetCacheEntryRequest.java [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+import java.util.Map;
+
+import static com.hazelcast.util.JsonUtil.getString;
+
+/**
+ * Request for fetching cache entries.
+ */
+public class GetCacheEntryRequest implements ConsoleRequest {
+
+    private final GetCacheEntryViewEntryProcessor entryProcessor = new GetCacheEntryViewEntryProcessor();
```
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/internal/management/request/GetCacheEntryRequest.java]: `static final`?

**Pull request merged**